src/modules/reservation.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable import/prefer-default-export */
+import { involvementApi, MoviesApi } from './utils.js';
+
+const popUp = document.createElement('div');
+popUp.className = 'reserve-pop-up';
+const footer = document.querySelector('footer');
+
+const addReservation = async (id, Username, startDate, endDate) => {
+  const reservation = {
+    item_id: id,
+    username: Username,
+    date_start: startDate,
+    date_end: endDate,
+  };
+  const response = await fetch(`${involvementApi}/reservations/`, {
+    method: 'POST',
+    body: JSON.stringify(reservation),
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+  return response;
+};
+
+const DisplayReservations = async (id, content) => {
+  const response = await fetch(`${involvementApi}/reservations?item_id=${id}`);
+  const result = await response.json();
+  result.reverse();
+  const resContainer = document.createElement('div');
+  result.forEach((data) => {
+    resContainer.innerHTML += `<p class='reservation-text'>${data.date_start} - ${data.date_end} by ${data.username}</p>`;
+  });
+  content.appendChild(resContainer);
+};
+const updateResDom = async (id, content, div) => {
+  const response = await fetch(`${involvementApi}/reservations?item_id=${id}`);
+  const result = await response.json();
+  const recent = result.length - 1;
+  const resText = document.createElement('p');
+  resText.className = 'reservation-text';
+  resText.innerHTML += `${result[recent].date_start} - ${result[recent].date_end} by ${result[recent].username}`;
+  content.prepend(resText);
+  div.textContent = `(${result.length})`;
+};
+export const reserveCounter = async (id, div) => {
+  const response = await fetch(`${involvementApi}/reservations?item_id=${id}`);
+  const result = await response.json();
+  const counter = result.length;
```

Comment:
According to the project requirements:

![image](https://user-images.githubusercontent.com/57421823/153610150-4ea837eb-425f-43b4-a826-2b3689a9ce5a.png)

- [ ] Therefore, kindly please make it so this counter does not use the API to give you the number directly.  

_Suggestion: You could just count the elements.  You might need to update the test._

That way, we will meet this requirement 💪

Reply:
Resolved ✔️